unity-client/Assets/Scripts/MainScripts/DCL/Controllers/BuilderMode/States/BuilderInWorldMode.cs

```diff
@@ -14,14 +14,24 @@ public class BuildMode : MonoBehaviour
 
     public float snapDistanceToActivateMovement = 10f;
 
+    [Header("Prefab references")]
+    public BuilderInWorldEntityHandler builderInWorldEntityHandler;
+    public ActionController actionController;
+    public BuilderInWorldBridge builderInWorldBridge;
+
     public System.Action OnInputDone;
-    public System.Action<BuildModeAction> OnActionGenerated;
+    public System.Action<BuildInWorldCompleteAction> OnActionGenerated;
```

Comment:
Please remember to add `event` modifier to all events to prevent using `Invoke` from outside of this class scope.

```suggestion
    public event System.Action<BuilderInWorldCompleteAction> OnActionGenerated;
```